src/main/java/com/auth0/net/BaseRequest.java

```diff
@@ -27,14 +28,12 @@
     @Override
     public T execute() throws Auth0Exception {
         okhttp3.Request request = createRequest();
-        Response response;
-        try {
-            response = client.newCall(request).execute();
+        try (Response response = client.newCall(request).execute()) {
+            return parseResponse(response);
+        } catch (APIException e) {
```

Comment:
Shouldn't this be `Auth0Exception` instead? Given that `APIException` is a subclass, people could later try/catch or ask what instanceof is the exception they receive and cast that to a `APIException` or read it as `Auth0Exception`. What are your thoughts?

Reply:
You're right, will do.

Reply:
Your comment got me thinking and you're very right. This not only is better for the user of this client, but it actually was a bug. If an exception of type `Auth0Exception` was thrown, it would pass through the `APIException` catch and be caught by the `IOException` catch where it got wrapped in an extra Auth0Exception with the message `Failed to execute request`. Changing the first catch to `Auth0Exception` fixes this. I realised this a little late, but in a second commit I improvement the tests a bit to prove this.